assets/src/scripts/components/hydrograph/index.js

```diff
@@ -107,6 +115,10 @@ class Hydrograph {
         //Redraw the current ts
         select('#ts-current')
             .attr('d', this.currentLine(this._tsData.current));
+        //remove the median stats plot and recreate
+        this.svg.selectAll('#median-point').remove();
+        this.svg.selectAll('#median-text').remove();
+        this._plotMedianPoints();
```

Comment:
When I merge this into my current branch, I'll have to modify this a bit - but I was wondering why we need the refresh here as well as in the constructor?

Reply:
The refresh happens here so that when the compare-timeseries is removed and the y-axis rescales, the median points will be re-positioned according to the updated y-axis. I was at a loss as to how to get that to happen more elegantly. :thinking:

Reply:
You might want to look at using the .call function. https://github.com/d3/d3-selection#selection_call

Reply:
Thanks for the suggestion.